levels.go

```diff
@@ -34,12 +37,147 @@ import (
 	"golang.org/x/time/rate"
 )
 
+const inactiveEpoch = math.MaxUint64
+
+type tableRefManager struct {
+	currentEpoch uint64
+
+	mu             sync.Mutex
+	guards         []*tableRef
+	inactiveGuards []*tableRef
+}
+
+func newTableRefManager(c *y.Closer) *tableRefManager {
+	e := new(tableRefManager)
+	go e.gcLoop(c)
+	return e
+}
+
+func (e *tableRefManager) refTables() *tableRef {
+	g := &tableRef{
+		epoch:      e,
+		localEpoch: atomic.LoadUint64(&e.currentEpoch),
+	}
+	e.mu.Lock()
+	e.guards = append(e.guards, g)
+	e.mu.Unlock()
+	return g
+}
+
+func (e *tableRefManager) gcLoop(c *y.Closer) {
+	ticker := time.NewTicker(10 * time.Millisecond)
+	defer ticker.Stop()
+	defer c.Done()
+
+	for {
+		select {
+		case <-ticker.C:
+			e.doGC()
+		case <-c.HasBeenClosed():
+			e.clear()
+			return
+		}
+	}
+}
+
+func (e *tableRefManager) clear() {
+	e.mu.Lock()
+	defer e.mu.Unlock()
+
+	for _, g := range e.inactiveGuards {
+		for _, d := range g.deletions {
+			for _, t := range d.tables {
+				t.Delete()
+			}
+		}
+	}
+
+	for _, g := range e.guards {
+		for _, d := range g.deletions {
+			for _, t := range d.tables {
+				t.Delete()
+			}
+		}
+	}
+}
+
+func (e *tableRefManager) doGC() {
+	currEpoch := atomic.LoadUint64(&e.currentEpoch)
+
+	e.mu.Lock()
+	inCurrentEpoch := true
+	gs := e.guards[:0]
+	for _, g := range e.guards {
+		localEpoch := atomic.LoadUint64(&g.localEpoch)
+		if localEpoch == inactiveEpoch {
+			e.inactiveGuards = append(e.inactiveGuards, g)
+		} else {
+			inCurrentEpoch = inCurrentEpoch && localEpoch == currEpoch
+			gs = append(gs, g)
+		}
+	}
+	e.guards = gs
+	e.mu.Unlock()
+
+	if !inCurrentEpoch {
+		return
+	}
+
+	prevEpoch := currEpoch - 1
```

Comment:
I think it's better to define an `epoch` type,  add methods like `load`, `incr`, `next`, `prev`